src/librustc_mir/transform/inline.rs

```diff
@@ -198,6 +168,45 @@ impl<'a, 'tcx> Inliner<'a, 'tcx> {
         }
     }
 
+    fn get_valid_function_call(&self,
+                               bb: BasicBlock,
+                               bb_data: &BasicBlockData<'tcx>,
+                               caller_mir: &Mir<'tcx>,
+                               param_env: ParamEnv<'tcx>,
+    ) -> Option<CallSite<'tcx>> {
+        // Don't inline calls that are in cleanup blocks.
+        if bb_data.is_cleanup { return None; }
+
+        // Only consider direct calls to functions
+        let terminator = bb_data.terminator();
+        if let TerminatorKind::Call { func: ref op, .. } = terminator.kind {
+            if let ty::FnDef(callee_def_id, substs) = op.ty(caller_mir, self.tcx).sty {
+                if let Some(instance) = Instance::resolve(self.tcx,
+                                                            param_env,
+                                                            callee_def_id,
+                                                            substs) {
+                    let is_virtual =
+                        if let InstanceDef::Virtual(..) = instance.def {
+                            true
```

Comment:
Since this is now its own function, you can get rid of `is_virtual` and just return `None` here.